lib/lita/handlers/artifactory.rb

```diff
@@ -92,6 +97,35 @@ def promote(response)
         response.reply reply_msg
       end
 
+      def push(response)
+        ruby_gem      = response.args[2]
+        version       = response.args[3]
+        human_name    = "#{ruby_gem} gem version #{version}"
+        gem_source    = "http://artifactory.chef.co/api/gems/gems-local/"
+        missing_gem   = false
+
+        Dir.mktmpdir do |dir|
+          Dir.chdir(dir) do
+            %w{ruby universal-mingw32}.each do |platform|
+              unless system(fetch_command_for_platform(platform, ruby_gem, version, gem_source))
+                response.reply("There were errors retrieving the #{human_name} from #{gem_source}!")
+                missing_gem = true
+              end
+            end
+
+            break if missing_gem
+
+            Dir.glob("*.gem") do |gem_file|
+              if system("gem push #{gem_file}")
```

Comment:
Per my comment on https://github.com/chef-cookbooks/oc-lita/pull/56/files#r73405888, I would use a non-standard keyname in your . You can then pass the `--key <NAME>` option to pick the key during the push.